lib/rpush/daemon/wns/toast_request.rb

```diff
@@ -22,16 +22,23 @@ class ToastRequestPayload
         def initialize(notification)
           @title = notification.data['title'] || ''
           @body = notification.data['body'] || ''
+          @launch = notification.data['launch']
+          @sound = notification.sound unless notification.sound.eql?("default".freeze)
         end
 
         def to_xml
-          "<toast>
+          launch_string = "" unless @launch
+          launch_string = " launch='#{CleanParamString.clean(@launch)}'" if @launch
```

Comment:
Yes, it is required. It is used a couple of lines below in:
```
"<toast#{launch_string}>
```

which is meant to produce either:
```xml
<toast>
```
or
```xml
<toast launch='...'>
```